src/uu/ls/src/ls.rs

```diff
@@ -1396,6 +1408,26 @@ fn display_items(items: &[PathData], config: &Config, out: &mut BufWriter<Stdout
     }
 }
 
+fn display_block_size(md: &Metadata, config: &Config) -> u64 {
+    /* GNU ls will display sizes in terms of block size
+       md.len() will differ from this value when the file has some holes
+    */
+    #[cfg(unix)]
+    {
+        // hard-coded for now - enabling setting this remains a TODO
+        let ls_block_size = 1024;
+        return match config.size_format {
+            SizeFormat::Binary => md.blocks() * 512,
+            SizeFormat::Decimal => md.blocks() * 512,
+            SizeFormat::Bytes => md.blocks() * 512 / ls_block_size,
+        };
+    }
+
+    #[cfg(not(unix))]
+    // unsupported for windows at the moment
+    0
```

Comment:
I think this should just return the size in bytes (through `md.blocks()` on Unix and `md.len()` on otherwise). The block size feels more like logic that should go into `display_file_size`. Then the name of this function could then be something like `get_block_size` or something like that.